src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py

```diff
@@ -485,6 +485,11 @@ def load_arguments(self, _):
             c.argument('protected_settings', type=validate_file_or_dict, help='Protected settings in JSON format for sensitive information like credentials. A JSON file path is also accepted.')
             c.argument('version', help='The version of the extension')
 
+            if scope == 'vm extension':
+                c.argument('extension_instance_name', help='Instance name of the extension. Default: name of the extension.', arg_group='Resource Id')
+            else:
+                c.argument('extension_instance_name', help='Instance name of the extension. Default: name of the extension.')
```

Comment:
This is a weird way to specify this and it duplicates stuf. It should look like this:
```Python
extension_instance_name_type = CLIArgumentType(help='Blah')

with self.argument_context('vm extension') as c:
  c.argument('extension_instance_name', extension_instance_name_type, arg_group='Resource id')

with self.argument_contenxt('vm extension blah blah') as c:
  c.argument('extension_instance_name', extension_instance_name_type)
```